core/src/test/java/org/apache/calcite/test/RelMetadataTest.java

```diff
@@ -678,6 +678,16 @@ private void checkRowCount(String sql, double expected, double expectedMin,
     checkRowCount(sql, 1D, 1D, 1D);
   }
 
+  @Test public void testRowCountAggregateConstantKey() {
+    final String sql = "select count(*) from emp group by 1";
+    checkRowCount(sql, 1.4D, 0D, 1D);
+  }
+
+  @Test public void testRowCountAggregateConstantKeys() {
+    final String sql = "select count(*) from emp group by 1,4";
+    checkRowCount(sql, 1.4D, 0D, 1D);
+  }
```

Comment:
For some sql engine, the group by 1 means group by the first field, we better change to other group by keys instead of just integer.

Reply:
@danny0405 This is a calcite test which I feel is descriptive enough to tell users that group by keys are (or will be considered) constant here.
I can add another test with non-integer keys but I believe we should keep this test as well.

Reply:
I agree. Calcite has a mode where "group by 1" means "group by first column" but it is not the default, and if it were, the test would fail.

Maybe add a test "select distinct 'x', 'y' from emp", which will turn into similar algebra.

Reply:
@julianhyde @danny0405 I have updated the existing tests to use what @julianhyde suggested